pandas/io/parsers.py

```diff
@@ -1620,6 +1620,12 @@ def __init__(self, src, **kwds):
 
         if self.usecols:
             usecols = _evaluate_usecols(self.usecols, self.orig_names)
+
+            #gh-14671
```

Comment:
@gfyoung is there a reason *some* of these checks for usecols are *outside* ``_evaluate_usecols``?

(e.g. the one after this one could be inside if ``self.names`` were passed)